elasticmagic-transport/src/commonMain/kotlin/dev/evo/elasticmagic/transport/TransportError.kt

```diff
@@ -2,25 +2,55 @@ package dev.evo.elasticmagic.transport
 
 import dev.evo.elasticmagic.serde.Deserializer
 import dev.evo.elasticmagic.serde.DeserializationException
+import dev.evo.elasticmagic.serde.forEach
 import dev.evo.elasticmagic.serde.toMap
 
 data class ErrorReason(
     val type: String,
     val reason: String,
     val resourceId: String?,
     val resourceType: String?,
+    val script: String?,
+    val scriptStack: List<String>,
+    val causedBy: CausedBy?,
 ) {
     companion object {
         fun parse(data: Deserializer.ObjectCtx): ErrorReason? {
             val type = data.stringOrNull("type") ?: return null
             val reason = data.stringOrNull("reason") ?: return null
             val resourceId = data.stringOrNull("resource.id")
             val resourceType = data.stringOrNull("resource.type")
+            val script = data.stringOrNull("script")
+            val scriptStack = buildList {
```

Comment:
Maybe rewrite it with a map
data.arrayOrNull("script_stack")?.map(String::toString)

Reply:
It's not a list. It's a `Deserializer.ArrayCtx` object and `forEach` is an extension method for it.